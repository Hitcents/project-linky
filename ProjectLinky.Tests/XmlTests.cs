using ApprovalTests;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.IO;
using System.Xml.Serialization;

namespace ProjectLinky.Tests
{
    [TestClass]
    public class XmlTests
    {
        [TestMethod]
        public void Serialize()
        {
            var serializer = new XmlSerializer(typeof(Config));

            using (var file = File.Create(TestConstants.TempFile))
            {
                serializer.Serialize(file, new Config
                {
                    Projects = new []
                    {
                        new Project
                        {
                            Path = "OurGame.iOS.csproj",
                            Rules = new[]
                            {
                                new Rule
                                {
                                    InputPattern = "Assets\\Images\\*.png",
                                },
                            },
                        },
                        new Project
                        {
                            Path = "OurGame.Droid.csproj",
                            Rules = new[]
                            {
                                new Rule
                                {
                                    InputPattern = "Assets\\Images\\*.png",
                                },
                            },
                        },
                        new Project
                        {
                            Path = "OurGame.WinRt.csproj",
                            Rules = new[]
                            {
                                new Rule
                                {
                                    InputPattern = "Assets\\Images\\*.png",
                                },
                            },
                        },
                    }
                });
            }

            Approvals.VerifyFile(TestConstants.TempFile);
        }
    }
}
