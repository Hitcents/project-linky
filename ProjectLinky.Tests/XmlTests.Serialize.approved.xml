<?xml version="1.0"?>
<linky xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <projects path="OurGame.iOS.csproj">
    <rules inputPattern="Assets\Images\*.png" />
  </projects>
  <projects path="OurGame.Droid.csproj">
    <rules inputPattern="Assets\Images\*.png" />
  </projects>
  <projects path="OurGame.WinRt.csproj">
    <rules inputPattern="Assets\Images\*.png" />
  </projects>
</linky>